tests/test_cases/test_cocotb/test_timing_triggers.py

```diff
@@ -110,91 +102,36 @@ async def test_timer_with_rational_units(dut):
     assert get_sim_time(units="fs") == time_fs + 1_000_000.0, "Expected a delay of 1 ns"
 
 
-exited = False
-
-
-async def do_test_readwrite_in_readonly(dut):
-    global exited
-    await RisingEdge(dut.clk)
-    await ReadOnly()
-    await ReadWrite()
-    exited = True
+@cocotb.test(expect_error=Exception)
+async def test_cached_write_in_readonly(dut):
+    """Test doing invalid sim operation"""
 
+    # only here because Riviera with VHPI looses its mind otherwise.
```

Comment:
Feel free to correct these or not.
```suggestion
    # only here because Riviera with VHPI loses its mind otherwise.
```